compliance/virtualmachine/relay_test.go

```diff
@@ -0,0 +1,144 @@
+package virtualmachine
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	v1 "github.com/stackrox/rox/generated/internalapi/virtualmachine/v1"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func TestVMRelay(t *testing.T) {
+	suite.Run(t, new(relayTestSuite))
+}
+
+type relayTestSuite struct {
+	suite.Suite
+
+	ctx context.Context
+}
+
+func (s *relayTestSuite) SetupTest() {
+	s.ctx = context.Background()
+}
+
+func (s *relayTestSuite) TestHandleVsockConnection_InjectsVsockCID() {
+	conn := s.defaultVsockConn().withVsockCID(42)
+	client := newMockSensorClient()
+
+	err := handleVsockConnection(s.ctx, conn, client)
+	s.Require().NoError(err)
+
+	s.Equal("42", client.capturedRequests[0].IndexReport.VsockCid)
+
+}
+
+func (s *relayTestSuite) TestHandleVsockConnection_RejectsMalformedData() {
+	conn := s.defaultVsockConn().withData([]byte("malformed-data"))
+	client := newMockSensorClient()
+
+	err := handleVsockConnection(s.ctx, conn, client)
+	s.Error(err)
+}
```

Comment:
Those two could be a single table-driven test with two cases.

Reply:
Maybe the next one `TestHandleVsockConnection_HandlesContextCancellation` as well?

Reply:
I didn't have that impression, I think they would fit nicely into a table-driven test. It may be a bit of work now, but then we could save the time in the future.

It is fully okay to do that in the followup in case you want to merge. (feel free to resolve this conversation then)